### PR TITLE
emulate hip/cuda-Memcpy3D with a kernel

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- add kernel to emulate hip/cuda-Memcpy3D
- add CMake option to enable/disable emulated memory copy (by default
only for HIP enabled)
- enable `ALPAKA_EMU_MEMCPY3D` for one HIP and one CUDA CI test

This optimization based on my [HIP issue](https://github.com/ROCm-Developer-Tools/HIP/issues/1815) and will increase the memory copy performance for device to device copies on the same device. 
I enabled the emulated copy only for HIP, for CUDA it can be optional enabled but is not showing any improvement. I assume the CUDA driver is already using a kernel instead of looping over the rows and call 1D mem-copies.

### CODE REVIEW

File: include/alpaka/mem/buf/uniformCudaHip/Copy.hpp [hunk 1]
```diff
@@ -53,6 +55,140 @@ namespace alpaka
...
                 {
+                    using vec3D = alpaka::vec::Vec<alpaka::dim::DimInt<3u>, size_t>;
+                    using vec2D = alpaka::vec::Vec<alpaka::dim::DimInt<2u>, size_t>;
+
+                    ///! copy 3D memory
+                    ///
+                    /// It is required to start  `height * depth` HIP/CUDA blocks.
+                    /// The kernel loops over the memory rows.
+                    template<typename T>
+                    __global__ void hipMemcpy3DEmulatedKernelD2D(
```
Comment (psychocoderHPC) [inline include/alpaka/mem/buf/uniformCudaHip/Copy.hpp:66]: note: a native kernel is used to avoid cyclic dependencies within alpaka.

File: cmake/alpakaCommon.cmake [hunk 2]
```diff
@@ -18,7 +19,10 @@ set(ALPAKA_ACC_CPU_B_SEQ_T_FIBERS_ENABLE_DEFAULT ON)
 set(ALPAKA_ACC_CPU_B_TBB_T_SEQ_ENABLE_DEFAULT ON)
 set(ALPAKA_ACC_CPU_B_OMP2_T_SEQ_ENABLE_DEFAULT ON)
 set(ALPAKA_ACC_CPU_B_SEQ_T_OMP2_ENABLE_DEFAULT ON)
-set(ALPAKA_ACC_CPU_BT_OMP4_ENABLE_DEFAULT ON)
+set(ALPAKA_ACC_ANY_BT_OMP5_ENABLE_DEFAULT ON)
+set(ALPAKA_ACC_ANY_BT_OACC_ENABLE_DEFAULT OFF)
```
Comment (BenjaminW3) [inline cmake/alpakaCommon.cmake]: I think this should not be part of this PR.

Comment (psychocoderHPC) [reply cmake/alpakaCommon.cmake]: fixed

**Review by sbastrakov — APPROVED** [review]

**Comment by psychocoderHPC** [comment]
@BenjaminW3 Do you know what we can do if a CI test fails. Actions do not allow to restart single tests.

**Comment by BenjaminW3** [comment]
Github is working on it. I do not know a solution despite rebuilding all or merging it nevertheless.

**Comment by psychocoderHPC** [comment]
Feel free to merge it even if the CI is not passing. The CI passed before but I fixed a indention issue in CMake and today not all tests passing.
